### PR TITLE
Added VSI driver handling via gdal:// scheme.

### PR DESCRIPTION
**Pull request by metasim**
Also added docs on scheme handling.

Depends on https://github.com/locationtech/rasterframes/pull/212

### CODE REVIEW

File: pyrasterframes/src/main/python/docs/raster-read.pymd [hunk 1]
```diff
@@ -51,19 +51,24 @@ display(tile)
...
-http:// ?
-s3a://
-s3n:// ?
-hdfs:// ?
-ftp:// ?
-azure ?
-google cloud ?
-
-gdal vsi? https://gdal.org/user/virtual_file_systems.html 
+RasterFrames relies on three different IO drivers, selected based on a combination of scheme, file extentions, and library availability. GDAL is used first if a compatible version of GDAL (>= 2.4) is installed. It is the only driver that can read non-GeoTIFF files. If GDAL is not available, either the _Java I/O` or _Hadoop_ driver will be selected, depending on scheme.
```
Comment (vpipkt) [inline pyrasterframes/src/main/python/docs/raster-read.pymd]: Possibly formatting issue here with Java I/O

File: pyrasterframes/src/main/python/docs/raster-read.pymd [hunk 2]
```diff
@@ -51,19 +51,24 @@ display(tile)
...
+
+| Prefix              | GDAL        | Java I/O | Hadoop |
+| ------------------- | ----------- | -------- | ------ |
+| `gdal://<vsidrv>//` |  ✔︎          | -︎        | -      |
+| `file://`           |  ✔︎          | ✔︎       | -      |
+| `http://`           |  ✔︎          | ✔︎       | -      |
+| `https://`          |  ✔︎          | ✔︎       | -      |
+| `ftp://`            | `/vsicurl/`  | ✔       | -      |
+| `s3://`             |  `/vsis3/` ︎  | ✔︎       | -      |
+| `hdfs://`           |  `/vsihdfs/` ︎| -       | ✔︎      |
```
Comment (vpipkt) [inline pyrasterframes/src/main/python/docs/raster-read.pymd]: maybe put hdfs above s3 and then s3, s3n, and s3a will all be together.

File: pyrasterframes/src/main/python/docs/raster-read.pymd (hunk 1 above)
Comment (metasim) [reply pyrasterframes/src/main/python/docs/raster-read.pymd]: What do you mean?

File: pyrasterframes/src/main/python/docs/raster-read.pymd [hunk 3]
```diff
@@ -51,19 +51,24 @@ display(tile)
...
+| `s3n://`            |  - ︎          | -        | ✔︎      |
+| `s3a://`            |  - ︎          | -   | ✔︎      |
+| `wasb://`           |  `/vsiaz/` ︎  | -       | ✔︎      |
+| `wasbs://`          |  - ︎          | -       | ✔︎      |
+
+Specific [GDAL Virtual File System drivers](https://gdal.org/user/virtual_file_systems.html) can be selected using the `gdal://<vsidrv>//` syntax. For example If you have a `archive.zip` file containing a GeoTiff named `my-file-inside.tif`, you can address it with `gdal://vsizip//path/to/archive.zip/my-file-inside.tif`. See the GDAL documentation for the format of the URIs after the `gdal:/` prefix (which is stripped off before passing the rest of the path to GDAL).
```
[the remark this answers is not in the document]
Comment (pomadchin) [reply pyrasterframes/src/main/python/docs/raster-read.pymd]: @metasim have you considered some synchronization with gt-contrib? [GDALDataPath](https://github.com/geotrellis/geotrellis-contrib/blob/master/gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALDataPath.scala
)

also we have SPI for all our `RasterSources` that look like (`gt+s3://`, `gdal+s3://`, etc)

**Pull request merged**